pycbc/results/pygrb_postprocessing_utils.py

```diff
@@ -804,3 +785,24 @@ def get_coinc_snr(trigs_or_injs, ifos):
     coinc_snr = numpy.sqrt(snr_sum_square)
 
     return coinc_snr
+
+
+def template_hash_to_id(trigger_file, bank_path):
+    """
+    This function converts the template hashes from a trigger file
+    into 'template_id's that represent indices of the
+    templates within the bank.
+    Parameters
+    ----------
+    trigger_file: h5py File object for trigger file
+    bank_file: filepath for template bank
+    """
+    bank = h5py.File(bank_path, "r")
+    hashes = bank['template_hash'][:]
+    ifos = [k for k in trigger_file.keys() if k != 'network']
+    trig_hashes = trigger_file[f'{ifos[0]}/template_hash'][:]
```

Comment:
Why is the `template_hash` dataset stored inside the ifo-specific groups, instead of the `network` group? Is that not unnecessarily duplicating the same dataset for each ifo?

Reply:
I'm not sure. The writing to file is done [here](https://github.com/gwastro/pycbc/blob/e2c35eff8367fa4700588a60f80737949d8fd254/pycbc/events/eventmgr.py#L787). The line that makes me unsure is [the definition of the index](https://github.com/gwastro/pycbc/blob/e2c35eff8367fa4700588a60f80737949d8fd254/pycbc/events/eventmgr.py#L719). It seems dependent on the ifo, but one would expect them to be the same.

Reply:
Yeah, what I see in `eventmgr.py` does not look correct, or at least not efficient. All template properties should be identical across interferometers. Maybe we do not need to worry about this in this PR, but it should be addressed shortly after this is merged. Please bring this up at the development call.

Reply:
I looked a bit more at the `eventmrg.py` code and the situation is more confusing than I thought, so I opened https://github.com/gwastro/pycbc/issues/4632 to track that issue.